[이린] step-1 : 1단계 구현 / 회원가입, 명단 정보 조회, 프로필 조회, html 중복코드 제거

new-pow opened the pull request:
## 변경사항
- 1단계 구현
- 월, 화 학습 내용 업데이트

## 고민했던 지점
- Lombok 사용에 대하여, 수정이 잦다보니 사용하지 않으면 Getter, Setter 수정하는데 너무 오래걸릴것 같아서 사용했다.
- `view` - `Controller` 데이터 주고받는게 어려웠으나 템플릿을 익힌 뒤 생각보다 쉽게 구현할 수 있었다.
- `gradle` Dependency 설정이 아직 많이 어렵다.
- 스프링에서 예외가 발생하면 어떻게 해야할 지 고민했으나 다음의 공부거리로 남겨두었다.
- 테스트 코드를 하는 것이 어려웠다.동료의 도움으로 기본 기능 테스트를 할 수 있었다.
- 동료와 함께 논의하며 구현을 하니 막힌 부분에서 쉽게 다음 단계로 갈 수 있었던 것 같다.

new-pow committed:
Setup (project) : 프로젝트 의존성 주입

- dev-tool
- lombok
- spring-jdbc

new-pow committed:
Add : step1을 위한 파일 생성

- 참고 링크를 기반으로 step1을 위한 파일 생성

new-pow committed:
Feat (User) : `User` 도메인 생성

- 회원가입 폼을 참고해 멤버 변수 설정

new-pow committed:
Refactor : 시그니처 통일 `Member` -> `User`

- 시그니처 통일을 위해 변수명 변경

new-pow committed:
Feat (UserRepository) : 미션 요구사항을 기반으로 메서드 작성

- 미션 요구사항인 회원가입, 회원 정보 보기, 회원 명단 보기를 위해 사용할 메서드 작성

new-pow committed:
Feat (UserService) : 미션 요구사항을 기반으로 메서드 작성

- 미션 요구사항인 회원가입, 회원 정보 보기, 회원 명단 보기를 위해 사용할 메서드 작성

new-pow committed:
Setup : lombok dependency 추가

- lombok 사용을 위해 dependency 추가

new-pow committed:
Setup : lombok dependency 추가

- lombok 사용을 위해 dependency 추가

new-pow committed:
Test (User) : 유저 서비스 테스트

- 기본 기능 테스트

new-pow committed:
Test (UserService) : 기능 테스트 생성자 수정

- 가입일 추가

new-pow committed:
Feat (User) : 도메인 멤버 수정

- 가입일 추가
- 시그니처 수정

new-pow committed:
Feat (dto) : `UserRequestDto` 생성

- 회원가입 시 입력받는 값 DTO 생성

new-pow committed:
Update (UserRepository) : 저장 객체 Map 구조에서 List로 변경

- 요구사항에 따라 수정

new-pow committed:
Fix (UserService) : 자료구조 변경으로 인해 객체 반환값 수정

- Optional<User> 로 반환되기 때문에 `.get()` 추가

new-pow committed:
Feat (UserController) : 1차 기능 구현(뷰페이지 반환, 회원가입, 회원 정보 조회 기능)

- 뷰페이지 반환 메서드
- 회원 가입 메서드 (dto 사용방식, 바로 객체 생성 방식)
- 회원 정보 조회 기능

new-pow committed:
Update (view) : html 문서 `templates` 로 이동 / `form` 수정 (method, action) / `list` mustache 로 요소를 출력할 수 있도록 수정

- html 문서 `templates` 로 이동
- `form` 수정 (method, action)
- `list` mustache 로 요소를 출력할 수 있도록 수정

new-pow committed:
Docs (README) : 3월 14일 학습목표, TODO 업데이트

1. `GET`, `POST`, `PUT` 등 메서드 방식에 대해 이해한다.
2. `mustache` 에 대해 공부하고 사용할 수 있다.
3. `MVC` 에 대해 이해한다.
4. 스프링부트의 테스트 방식을 이해한다.

new-pow committed:
Setup (build) : mustache dependency 추가

- 템플릿 사용을 위한 dependency 추가

new-pow committed:
Update (User) : entity 멤버 수정

- 사용하지 않는 멤버 삭제

new-pow committed:
Update (dto) : `User` 멤버 수정에 따른 dto 수정

- 변수명 수정

new-pow committed:
Feat (dto) : `userId` 를 생성하는 메서드 구현

- 현재 일자를 해쉬코드로 바꾸어 저장

new-pow committed:
Feat (CafeController) : index view method

- 동적으로 변경

new-pow committed:
Feat (UserController) : 매핑 url 수정 / Repository 를 Service로 변경

- 매핑 url을 요구사항에 맞게 수정
- repository 멤버를 Service 멤버로 변경

new-pow committed:
Feat (UserService) : 반환값이 Null일 때 예외처리

- 유저 1개를 조회할 때 Null 값 반환시 예외발생

new-pow committed:
Refactor : 사용하지 않는 멤버, 라인 삭제

- 사용하지 않는 멤버와 코드 라인 삭제

new-pow committed:
Test (UserServiceTest) : 기능 테스트

- service 기능 테스트

new-pow committed:
Feat (User) : `userId`를 비교해주는 메서드 구현

- isIdEquals 구현

new-pow committed:
Refactor (UserRepository) : `userId`를 비교해주는 메서드 적용

- findById 메서드 리팩토링

new-pow committed:
Todo : 할일 주석 추가

- 중복 userId 처리

new-pow committed:
Feat (exception) : 중복 이름 입력시 예외처리

- 스프링 예외처리를 더 공부해서 어떻게 구현할 수 있는지 개선해봐야겠다.

new-pow committed:
Feat (User) : 같은 `nickname` 이 있을 경우 예외 발생

- 스프링 예외처리를 공부해서 개선할 예정

new-pow committed:
Feat (User) : 같은 이름이 있으면 true를 반환하는 함수 구현

- 같은 이름이 있을 경우, 예외처리를 하는 데 사용할 예정

new-pow committed:
Update (RequestDto) : 회원 코드를 생성하는 로직 수정

- 중복 `userId` 방지를 위해 1씩 커지도록 수정

new-pow committed:
Feat (html) : 회원 프로필 조회를 위한 페이지 설정

- mustache를 이용하여 변수 설정

new-pow committed:
Feat (view) : html 중복 코드 제거

- footer, header, nav 로 나누어 중복코드 제거

new-pow committed:
Refactor : 사용하지 않는 `dependency` 삭제

- 중복 역할을 하는 것 같아 삭제함

new-pow committed:
Feat (UserListResponseDto) : 유저 목록 볼 때 반환값 DTO 구현

- `user/list` 요청시 반환 데이터 dto

new-pow committed:
Feat (UserController) : user 반환하는 요청에 대해 dto로 반환값 수정

- user 반환하는 요청에 대해 dto로 반환값 수정

new-pow committed:
Fix (UserJoinRequestDto) : 회원코드 생성 로직 변경

- 회원코드가 중복되어 생성되는 에러가 있어 로직 변경

new-pow committed:
Refactor (User) : Dto가 가진 회원 코드 생성 기능을 User로 이전

- DTO가 로직을 가지는 것이 어색해서 user에게 옮김

new-pow committed:
Refactor : 사용하지 않는 코드 삭제

- 사용하지 않는 코드 삭제, 잘못 삭제된 html 코드 추가

new-pow committed:
Refactor : 시그니처 오타 수정

- 오타수정

new-pow committed:
Dcos (README) : 금일 학습 내용 업데이트

- 학습내용 업데이트

honux77 commented on src/main/java/kr/codesqaud/cafe/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,37 @@
+import java.util.List;
+import java.util.NoSuchElementException;
+
+@AllArgsConstructor
```
프로젝트 기간 전까지 학습 기간에 롬복은 사용하지 않습니다. 다음 단계 하기 전에 롬복 제거해 주세요.

honux77 commented on src/main/java/kr/codesqaud/cafe/controller/UserController.java:
```diff
@@ -0,0 +1,50 @@
+
+    private UserService userService;
+
+    @GetMapping("/user/form")
+    public String addForm() {
+        return "user/form";
+    }
```
이런건 static 파일을 사용하셔도 됩니다.

honux77 commented on src/main/java/kr/codesqaud/cafe/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Override
+    public User findUser(Long id) {
+        if (repository.findById(id).isEmpty()) {
+            throw new NoSuchElementException();
+        }
+        return repository.findById(id).get();
```
이 부분 더 좋은 방법은 없는지 고민해 보세요.

honux77 reviewed (changes requested):
롬복 제거해서 다시 push 해 주세요.

new-pow committed:
Refactor : lombok 제거

- 롬복 제거
- 생성자 수정

honux77 commented:
@new-pow 직접 머지 가능하면 머지하고 안 되면 DM 보내 주세요.

new-pow committed:
Fix : 의존성 주입 다시 해주었다.

- lombok을 정리하면서 의존성 주입을 다시 해주었다.

new-pow committed:
Docs (README) : 3월 15일 학습 목표 재설정

- 금일의 목표
1. 전날에 학습하지 못한 부분을 학습한다.
2. 의존성 주입의 의미와 방식에 대해 이해해본다.
3. `MVC` 에 대해 이해한다.
